### PR TITLE
adding retry event count metric for source

### PR DESCRIPTION
**Pull request by kthakar1990**
# Changes
- adding a code change that will be used by `knative/eventing` to report retry event count for knative source.

<!--
In addition, categorize the changes you're making using the "/kind" Prow command, example:

/kind enhancement

Supported kinds are: api-change, bug, cleanup, deprecation, removal, documentation, enhancement, performance

### CODE REVIEW

**Comment by kthakar1990** [comment]
Hey guys, I will take care of the CLA.. I have signed CLA for knative/eventing and knative/sandbox but looks like missed it for this one.. In the mean time, please review the PR and let me know if there are any updates that's needed.

File: source/source_stats_reporter.go [hunk 1]
```diff
@@ -98,6 +108,16 @@ func (r *reporter) ReportEventCount(args *ReportArgs, responseCode int) error {
 	return nil
 }
 
+func (r *reporter) ReportRetryEventCount(args *ReportArgs, responseCode int) error {
+	fmt.Print("Report retry_event_count Successfully called!")
```
Comment (lionelvillard) [inline source/source_stats_reporter.go:112]: no print statement. Can you please remove it.

File: source/source_stats_reporter.go [hunk 2]
```diff
@@ -90,6 +99,7 @@ func NewStatsReporter() (StatsReporter, error) {
 }
 
 func (r *reporter) ReportEventCount(args *ReportArgs, responseCode int) error {
+	fmt.Print("Report event_count Successfully called!")
```
Comment (lionelvillard) [inline source/source_stats_reporter.go:102]: same here, no print

**Review by vagababov — COMMENTED** [review]
Any reason all those are in `pkg` rather than in `eventing`?

**Comment by kthakar1990** [comment]
@googlebot I signed it!

**Comment by kthakar1990** [comment]
> Any reason all those are in `pkg` rather than in `eventing`?

There will be another PR in eventing repo which is dependent on this PR getting merged.. For now I am just adding a method in source reporter. This repo is part of vendor directory in eventing repo and these changes are essential for reporting metrics.

**Comment by lionelvillard** [comment]
> Any reason all those are in `pkg` rather than in `eventing`?

Just a historical reason I believe. I agree this package should belong to eventing, not pkg.

**Comment by kthakar1990** [comment]
https://github.com/knative/pkg/pull/2016 moved it to a new PR!

**Pull request closed without merging**